src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderTest.cs

```diff
@@ -329,6 +330,207 @@ public static void CheckLegacyNullRowVersionIsEmptyArray()
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static async void ExecuteReader_CommandBehavior_Default_Async_Test2()
+        {
+            using SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString);
+            connection.Open();
+
+            using var cmd = connection.CreateCommand();
+            cmd.CommandText = "SELECT 'abcdefg', 'hijklmn'";
+            using var reader = cmd.ExecuteReader(CommandBehavior.Default);
+            {
+                reader.Read();
+                using var tr1 = reader.GetTextReader(0);
+                {
+                    reader.GetValue(1);
+
+                    char[] value = new char[1];
+                    int value2 = await tr1.ReadAsync(value, 0, 1);
+
+                    Assert.True(value2 > -1, "Expected to read a value successfully.");
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static void ExecuteReader_CommandBehavior_Default_Test3()
+        {
+            using SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString);
+            connection.Open();
+
+            using var cmd = connection.CreateCommand();
+            cmd.CommandText = "SELECT 'abcdefg', 'hijklmn'";
+            using var reader = cmd.ExecuteReader(CommandBehavior.Default);
+            {
+                reader.Read();
+                using var tr1 = reader.GetTextReader(0);
+                using var tr2 = reader.GetTextReader(1);
+
+                tr2.Read();
+                int value2 = tr1.Read();
+                Assert.True(value2 > -1, "Expected to read a value successfully.");
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static void ExecuteReader_CommandBehavior_Default_Async_Test4()
+        {
+            using SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString);
+            connection.Open();
+
+            using var cmd = connection.CreateCommand();
+            cmd.CommandText = "SELECT 'abcdefg', 'hijklmn'";
+            using var reader = cmd.ExecuteReader(CommandBehavior.Default);
+            {
+                reader.Read();
+                using var tr1 = reader.GetTextReader(0);
+                using var tr2 = reader.GetTextReader(1);
+
+                char[] value2 = new char[1];
+                tr2.ReadAsync(value2, 0 ,1);
+                char[] value1 = new char[1];
+                tr1.ReadAsync(value1, 0, 1);
+                Assert.True(value1[0] > -1, "Expected to read a value successfully.");
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static void ExecuteReader_CommandBehavior_SequentialAccess_Test5()
```

Comment:
Maybe consider 
`ExecuteReader_WhenCommandBehaviorIsSequentialAccess_ValueIsNotReadSequentiallyAndThrowsObjectDisposedException`

Reply:
Method name renamed to one that conforms to the test naming convention:

The name of your test should consist of three parts:

The name of the method being tested.
The scenario under which it's being tested.
The expected behavior when the scenario is invoked.